src/main/java/core/basesyntax/interfaces/WriteToFileImpl.java

```
@@ -0,0 +1,21 @@
+package core.basesyntax.interfaces;
+
+import java.io.BufferedWriter;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class WriteToFileImpl implements WriteToFile {
+
+    @Override
+    public void write(String filePath, List<String> report) {
```

Add null/argument validation at the start of the method. The checklist requires handling null parameters and throwing informative RuntimeExceptions rather than allowing NPEs: check filePath is non-null/non-empty and report is non-null and throw RuntimeException with context if invalid .